### PR TITLE
CSI 5437 selector check in predicate

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

File: controllers/utils/predicate.go [hunk 1]
```diff
@@ -54,16 +56,21 @@ func CreateRequests(client runtimeclient.Client) handler.EventHandler {
...
-			requests := make([]ctrl.Request, len(vgList.Items))
+			var requests []ctrl.Request
 			for _, vg := range vgList.Items {
-				requests = append(requests, ctrl.Request{
-					NamespacedName: types.NamespacedName{
-						Namespace: vg.Namespace,
-						Name:      vg.Name,
-					},
-				})
+				selector, err := metav1.LabelSelectorAsSelector(vg.Spec.Source.Selector)
```
Comment (matancarmeli7) [inline controllers/utils/predicate.go:62]: will it work on VGs without a selector field/Source field?

File: controllers/utils/predicate.go [hunk 2]
```diff
@@ -54,16 +56,21 @@ func CreateRequests(client runtimeclient.Client) handler.EventHandler {
...
+				if err != nil {
+					return []ctrl.Request{}
+				}
+				if selector.Matches(labels.Set(object.GetLabels())) {
```
Comment (matancarmeli7) [inline controllers/utils/predicate.go:66]: can you try and use [this](https://github.com/IBM/csi-volume-group-operator/blob/develop/controllers/utils/labels.go#L35-L38) function?

File: controllers/utils/predicate.go (hunk 1 above)
Comment (ArbelNathan) [reply controllers/utils/predicate.go:62]: selector is mandatory field..

File: controllers/utils/predicate.go (hunk 2 above)
Comment (ArbelNathan) [reply controllers/utils/predicate.go]: done ( used areLabelsMatchLabelSelector)

**Review by matancarmeli7 — APPROVED** [review]
LGTM

**Pull request merged**